### ISSUE TITLE
Support for Propagate Default Route in azurerm_virtual_hub_connection

### ISSUE THREAD

**Issue by shin-akuma** [terraform-providers/terraform-provider-azurerm#12742]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

There should be an option to enable or disable Propagate Default Route when creating a vnet connection in virtual hub

### New or Affected Resource(s)

* azurerm_virtual_hub_connection

### Potential Terraform Configuration

propagate_default_route - (Optional) Allows Virtual Hub to propagate a learnt default route to this connection. E.g. 0.0.0.0/0 with next hop to firewall. Defaults to false.

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

**Comment by aidapsibr** [comment]
Is this the `internet_security_enabled` option? I noticed my plan changed to true when I manually enabled and can't find that setting in the portal. I'm assuming they are one and the same.

**Comment by jackperrypresidio** [comment]
I don't believe this issue has been fixed yet.  Hopefully someone from Hashicorp will see this, and this gets resolved soon.

> ### Community Note
> * Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
> * Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
> * If you are interested in working on this issue or have submitted a pull request, please leave a comment
> 
> ### Description
> There should be an option to enable or disable Propagate Default Route when creating a vnet connection in virtual hub
> 
> ### New or Affected Resource(s)
> * azurerm_virtual_hub_connection
> 
> ### Potential Terraform Configuration
> propagate_default_route - (Optional) Allows Virtual Hub to propagate a learnt default route to this connection. E.g. 0.0.0.0/0 with next hop to firewall. Defaults to false.
> 
> ```terraform
> # Copy-paste your Terraform configurations here - for large Terraform configs,
> # please use a service like Dropbox and share a link to the ZIP file. For
> # security, you can also encrypt the files using our GPG public key.
> ```
> 
> ### References
> * #0000

This issue has not been resolved yet.  Hopefully, someone from Hashicorp sees this, and they resolve this soon!

**Comment by bizrad** [comment]
Setting `internet_security_enabled = true` in the `azurerm_virtual_hub_connection` caused this setting to show enabled in the UI for me. I'm also using the vWAN hub premium firewall, not sure if that's related.
